edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -155,3 +182,176 @@ def test_has_permission(self, enforce_scopes, authentication_class, jwt_filters,
+
+    def test_anonymous_fails(self):
+        request = self._create_request()
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, 401)
+
+    def test_session_staff_succeeds(self):
+        user = self._create_user(is_staff=True)
+        request = self._create_request()
+        self._create_session(request, user)
+
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, 200)
+
+    @patch('edx_rest_framework_extensions.permissions.log')
+    def test_session_user_not_in_url_fails(self, mock_log):
+        user = self._create_user()
+        request = self._create_request()
+        self._create_session(request, user)
+        
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, 403)
+        self._assert_log(mock_log, self.IsUserInUrlLog)
+
+    def test_session_user_in_url_succeeds(self):
+        user = self._create_user()
+        request = self._create_request(username_in_url=user.username)
+        self._create_session(request, user)
+
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, 200)
+
+    JwtTestCase = namedtuple('JwtTestCase', 'is_enforced, is_restricted, is_user_in_url, expected_response, expected_log')
+
+    @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
+    @patch('edx_rest_framework_extensions.permissions.log')
+    @ddt.data(
+        # **** Enforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=True, expected_response=403, expected_log=JwtHasScopeLog),
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=JwtHasScopeLog),
+
+        # **** Unenforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+    )
+    @ddt.unpack
+    def test_jwt_cases(self, is_enforced, is_restricted, is_user_in_url, expected_response, expected_log, mock_log, waffle_mock):
```

Comment:
Is this `test_jwt_cases_without_scopes_and_filters`? Other?  I think better names would make it more clear if we were missing coverage.

Reply:
do this.

Reply:
done.